### PR TITLE
CCTRI-1780, CCTRI-1839, CCTRI-1786: Add authentication errors handling and tests for them, logging, /vers…

### PR DESCRIPTION
**Pull request by MaxMhonc**
…ion endpoint.

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -1,8 +1,11 @@
 from flask import Flask, jsonify
 
 from api.enrich import enrich_api
+from api.errors import TRFormattedError
 from api.health import health_api
 from api.respond import respond_api
+from api.utils import jsonify_data
+from api.version import version_api
```
Comment (GeorgiyII) [inline app.py:8]: please swap this imports, endpoints should be together and one by one

File: app.py [hunk 2]
```diff
@@ -12,10 +15,13 @@
 app.register_blueprint(health_api)
 app.register_blueprint(enrich_api)
 app.register_blueprint(respond_api)
+app.register_blueprint(version_api)
 
 
 @app.errorhandler(Exception)
 def handle_error(exception):
+    app.logger.error(exception)
+
```
Comment (GeorgiyII) [inline app.py:24]: remove empty row please

**Review by GeorgiyII — APPROVED** [review]

File: api/utils.py [hunk 3]
```diff
@@ -1,7 +1,29 @@
...
-from authlib.jose.errors import JoseError
+from authlib.jose.errors import DecodeError, BadSignatureError
 from flask import request, current_app, jsonify
-from werkzeug.exceptions import Forbidden, BadRequest
+from werkzeug.exceptions import BadRequest
+
+from api.errors import AuthorizationError
+
+
+def gre_auth_token():
```
Comment (AlinaBlahun) [inline api/utils.py:9]: `get` not `gre`

File: tests/unit/api/test_enrich.py [hunk 4]
```diff
@@ -33,7 +28,7 @@ def test_enrich_call_with_valid_jwt_but_invalid_json_failure(route,
     response = client.post(route,
                            headers=headers(valid_jwt),
                            json=invalid_json)
-    assert response.status_code == HTTPStatus.BAD_REQUEST
+    assert response.status_code == HTTPStatus.OK
```
Comment (AlinaBlahun) [inline tests/unit/api/test_enrich.py:31]: Please, check  that body has an error here

File: api/utils.py (hunk 3 above)
Comment (AlinaBlahun) [inline api/utils.py:9]: get not gre

File: tests/unit/api/test_enrich.py (hunk 4 above)
Comment (AlinaBlahun) [inline tests/unit/api/test_enrich.py:31]: Please, check that body has an error here

**Review by AlinaBlahun — APPROVED** [review]

**Pull request merged**